lib/acts_as_taggable_on/tag.rb

```diff
@@ -10,7 +10,7 @@ class Tag < ::ActiveRecord::Base
     ### VALIDATIONS:
 
     validates_presence_of :name
-    validates_uniqueness_of :name, if: :validates_name_uniqueness?
+    validates_uniqueness_of :name, if: :validates_name_uniqueness?, case_sensitive: true
```

Comment:
This would have to use ActsAsTaggableOn.strict_case_match rather than a hard-coded true.